tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,278 @@
+				cfg.Global.InsecureFlag, cfg.Global.ClusterID, cfg.Global.VCenterHostname, cfg.Global.User, cfg.Global.Password, cfg.Global.Datacenters, cfg.Global.VCenterPort)
+
+			// Set modified conf file and update
+			currentSecret.Data[csivsphereConf] = []byte(modifiedConf)
+			_, err := client.CoreV1().Secrets(kubeSystemNamespace).Update(ctx, currentSecret, metav1.UpdateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			time.Sleep(time.Duration(90) * time.Second)
+		}
+	})
+
+	/*
+		Test to verify cluster distribution set in PVC is being honored during volume creation.
+		Steps
+			1. Create StorageClass
+			2. Create PVC with valid disk size
+			3. Expect PVC to pass and cluster distribution value set
+	*/
+	ginkgo.It("Verify dynamic provisioning of pv has cluster-distribution value set", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Invoking cluster-distribution for volume basic test")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+
+		scParameters[scParamDatastoreURL] = datastoreURL
+		storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, pvclaim.Namespace, pvclaim.Name, framework.Poll, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", volHandle))
+		queryResult, err := e2eVSphere.queryCNSVolumeWithResult(volHandle)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(queryResult.Volumes) > 0)
+
+		framework.Logf("Cluster-distribution value on CNS is %s \n", queryResult.Volumes[0].Metadata.ContainerCluster.ClusterDistribution)
+
+		if queryResult.Volumes[0].Metadata.ContainerCluster.ClusterDistribution != "CSI-Vanilla" {
+			err = fmt.Errorf("Wrong cluster-distribution name present on CNS")
```

Comment:
Wrong/missing - might be better in the error message
Also, `gomega.Expect(queryResult.Volumes[0].Metadata.ContainerCluster.ClusterDistribution).Should(gomega.Equal("CSI-Vanilla"), <err_msg>)` might be better